utils/variables_computation/jet_variables.py

```diff
@@ -1,5 +1,100 @@
 import numpy as np
 import awkward as ak
+import utils.awkward_array_utilities as akUtl
+import numba as nb
+from utils.Logger import *
+
+NAN_VALUE = -1
+
+
+def delta_phi(obj1, obj2):
+    return obj1.delta_phi(obj2)
+
+def delta_rapidity(obj1, obj2):
+    return obj1.rapidity - obj2.rapidity
+
+def delta_r_rapidity(obj1, obj2):
+    
+    dy = delta_rapidity(obj1, obj2)
+    dphi = delta_phi(obj1, obj2)
+
+    return np.sqrt(dy**2 + dphi**2)
+
+def delta_r_pseudorapidity(obj1, obj2):
+    
+    return obj1.delta_r(obj2)
+
+
+def delta_r(obj1, obj2, use_rapidity=False):
+    if use_rapidity:
+        return delta_r_rapidity(obj1, obj2)
+    else:
+        return delta_r_pseudorapidity(obj1, obj2)
+
+
+
+def calculate_generalized_angularity(constituents, jets, jet_radius, beta, kappa, nan_value=NAN_VALUE):
+    """Calculate generalized angularity.
+
+    Args:
+        constituents (awkward.Array or None):
+            2D ak array of constituents per jet, where axis 0 is the jet axis,
+            axis 1 is the constituents axis with fields pt, eta, phi, mass,
+            and with name PtEtaPhiMLorentzVector.
+            Can be None if kappa=0.
+        jets (awkward.Array or None):
+            1D ak array of jets, with fields pt, eta, phi and mass and with
+            name PtEtaPhiMLorentzVector.
+            Can be None if beta=0.
+        beta (float): Angular distance exponent.
+        kappa (float): Transverse momentum fraction exponent.
+        nan_value (float, optional):
+            Value to use when the angularity cannot be computed (should not
+            happen!)
+
+    Returns:
+        awkward.Array
+
+    Examples:
+        >>> from tests.utilities.variablesComputation.awkwardArray.docTestHelper import get_data
+        >>> from builtinUtilities import pretty_printer
+        >>> constituents, jets = get_data()
+        >>> pretty_printer(calculate_generalized_angularity(constituents, jets, 1, 0, 2))
+        [0.500, -1.000, 0.297]
+        >>> pretty_printer(calculate_generalized_angularity(constituents, jets, 1, 1, 1))
+        [0.071, -1.000, 0.168]
+    """
+
+    if beta == 0 and kappa == 0:
+        generalized_angularity = ak.num(constituents.pt, axis=1)
+
+    else:
+        if beta == 0:
+            angular_term = 1.
+        else:
+            jet_broadcasted = akUtl.broadcast(jets, constituents)[0]
+            delta_r = delta_r(constituents, jet_broadcasted)
```

Comment:
Here you can just do `constituents.delta_r(jet_broadcasted)` and remove the 5 small functions above :)

Reply:
done